packages/aws-cdk-lib/aws-autoscaling/lib/auto-scaling-group.ts

```diff
@@ -686,6 +686,42 @@ export interface AutoScalingGroupProps extends CommonAutoScalingGroupProps {
    * @default false
    */
   readonly requireImdsv2?: boolean;
+
+  /**
+   * Specifies the upper threshold as a percentage of the desired capacity of the Auto Scaling group.
+   * It represents the maximum percentage of the group that can be in service and healthy, or pending,
+   * to support your workload when replacing instances.
+   *
+   * Value range is 0 to 100. After it's set, both `maintenancePolicyMinHealthPercentage` and
+   * `maintenancePolicyMaxHealthPercentage` to -1 will clear the previously set value.
+   *
+   * Both or neither of `maintenancePolicyMinHealthPercentage` and `maintenancePolicyMaxHealthPercentage`
+   * must be specified, and the difference between them cannot be greater than 100. A large range increases
+   * the number of instances that can be replaced at the same time.
+   *
+   * @see https://docs.aws.amazon.com/autoscaling/ec2/userguide/ec2-auto-scaling-instance-maintenance-policy.html
+   *
+   * @default - No instance maintenance policy.
```

Comment:
If possible, I think we should just set the default to `-1`. If I am reading the docs correctly, this is equivalent to undefined in CloudFormation. The only issue I see is if CloudFormation somehow fails if you set the value to `-1` and there wasn't any previous value to clear. If that happens, then lets go down this path. If not, we can treat `-1` and `undefined` equivalently in CDK.

Reply:
When I tried to run CloudFormation with -1 from the beginning, I got the following error in CFn. In other words, it seems that -1 and undefined must be distinguished.

```
MinHealthyPercentage must be between 0 and 100 (inclusive).
```

Reply:
:(